### DISCUSSION TITLE
SyntaxError: Unexpected end of JSON input at JSON.parse for GET method

### DISCUSSION THREAD

**Discussion by shivam-taneja** [vercel/next.js#56890] [Help] [answered]
### Summary

I've read this https://github.com/vercel/next.js/discussions/51011 discussion regarding the issue i'm facing but still can't figure out a solution for why `request.json()` doesn't work for a GET method. Error is caused by `await request.json();` part.

```
import { NextRequest, NextResponse } from "next/server";

export async function GET(request: NextRequest) {

  const body = await request.json();
  console.log(body);

  return NextResponse.json('nice');
}
```
![image](https://github.com/vercel/next.js/assets/79853285/a2fd2cc5-399d-449f-b8ab-9e0b3d73e307)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

The problem is that the body is not copied... 

You can see it here:

https://github.com/vercel/next.js/blob/canary/packages/next/src/server/web/sandbox/sandbox.ts#L95-L97

```ts
 const cloned = !['HEAD', 'GET'].includes(params.request.method)
    ? params.request.body?.cloneBodyStream()
    : undefined
```

And then later on:

```ts
    const result = await edgeFunction({
      request: {
        ...params.request,
        body:
          cloned && requestToBodyStream(runtime.context, KUint8Array, cloned),
      },
    })
```

As said in #51011, whether GET should allow a body in the request is murky waters.

**Reply by shivam-taneja** [reply]
Ok thanks!

Just one more question, what do you suggest I should do now? since GET method won't work.

**Reply by icyJoseph** [reply]
What is the use case? Is it an API call not controlled by you?

**Reply by shivam-taneja** [reply]
I want to implement a basic GET method in which I'm sending the user ID in the body, and in response I'm getting the data for that user if that user exists in the DB.

**Reply by icyJoseph** [reply]
Then you probably should use POST instead. Or use URL query parameters instead.
